kivy/uix/behaviors/compoundselection.py

```diff
@@ -308,7 +308,7 @@ def select_with_touch(self, node, touch=None):
             return False
 
         if (node in self.selected_nodes and (not range_select)):  # selected
-            if multiselect:
+            if ((not multi) or multiselect):
                 self.deselect_node(node)
             else:
                 self.clear_selection()
```

Comment:
@dessant  I think its already been taken care of since line 315 is executed only when self.multiselect=True and touch_multiselect as well as ctrl_down are False and that is my use case. So no need to put an if I guess.